src/datadog_variable.cpp

```diff
@@ -52,16 +48,29 @@ static ngx_int_t expand_span_variable(ngx_http_request_t* request,
     return NGX_OK;
   }
 
-  auto span_variable_value =
-      context->lookup_span_variable_value(request, suffix);
-  variable_value->len = span_variable_value.len;
-  variable_value->valid = true;
-  variable_value->no_cacheable = true;
-  variable_value->not_found = false;
-  variable_value->data = span_variable_value.data;
+  auto set_variable = [&](std::string_view suffix) {
+    auto span_variable_value =
+        context->lookup_span_variable_value(request, suffix);
+    variable_value->len = span_variable_value.len;
+    variable_value->valid = true;
+    variable_value->no_cacheable = true;
+    variable_value->not_found = false;
+    variable_value->data = span_variable_value.data;
+  };
+
+  auto variable_name = to_string_view(*reinterpret_cast<ngx_str_t *>(data));
+  if (variable_name.starts_with("datadog_")) {
+    auto suffix =
+        slice(variable_name, TracingLibrary::span_variables().prefix.size());
+    set_variable(suffix);
+  } else if (std::string_view otel("opentelemetry_");
```

Comment:
Above we are not using a string view. Can we make the two cases consistent ?

Reply:
Ok, we can use prefix.size(). So it's a nit comment (I'm fine with staying like that)